project/util/util.go

```diff
@@ -0,0 +1,193 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	cliconfig "github.com/docker/cli/cli/config"
+	"github.com/docker/cli/project"
+)
+
+const (
+	recentProjectsFileName = ".recentProjects.json"
+)
+
+// ChRootDir changes the current working directory to project's root directory
+func ChRootDir(p project.Project) (previousWorkDir string, err error) {
+	previousWorkDir, err = os.Getwd()
+	if err != nil {
+		return
+	}
+	err = os.Chdir(p.RootDir())
+	if err != nil {
+		return
+	}
+	return
+}
+
+type recentProject struct {
+	IDVal      string `json:"id"`
+	RootDirVal string `json:"root"`
+	Timestamp  int    `json:"t"`
+}
+
+// Project interface implementation
+func (rp *recentProject) RootDir() string {
+	return rp.RootDirVal
+}
+func (rp *recentProject) ID() string {
+	return rp.IDVal
+}
+func (rp *recentProject) Dial() (net.Conn, error) {
```

Comment:
Are those 2 methods required here ? (also, could be elsewhere than in `util` package, `project.Recents` might make senses for example)